src/sms_verification/repository.rs

```diff
@@ -0,0 +1,208 @@
+use crate::infrastructure::database::DbError;
+use crate::infrastructure::database::SqlDb;
+use chrono::{DateTime, Utc};
+use sea_query::{Expr, PostgresQueryBuilder, Query};
+use sea_query_binder::SqlxBinder;
+use sqlx::Row;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum SmsVerificationRepositoryError {
+    #[error("SMS verification not found: {0}")]
+    NotFound(String),
+
+    #[error("No active verification session for phone number: {0}")]
+    NoActiveVerification(String),
+
+    #[error("{0}")]
+    DatabaseError(#[from] DbError),
+
+    #[error("Database error: {0}")]
+    Database(#[from] sqlx::Error),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, sqlx::Type)]
+#[sqlx(type_name = "text")]
+pub enum VerificationStatus {
+    #[sqlx(rename = "PENDING")]
+    Pending,
+    #[sqlx(rename = "VERIFIED")]
+    Verified,
+    #[sqlx(rename = "FAILED")]
+    Failed,
+}
+
+impl VerificationStatus {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            VerificationStatus::Pending => "PENDING",
+            VerificationStatus::Verified => "VERIFIED",
+            VerificationStatus::Failed => "FAILED",
+        }
+    }
+}
+
+#[derive(Debug, Clone, sqlx::FromRow)]
+#[allow(dead_code)]
+pub struct SmsVerification {
+    pub unique_id: i32,
```

Comment:
Just id assuming this is the database id?
```suggestion
    pub id: i32,
```